### PR TITLE
wal: release wal locks before renaming directory on init

### LINKED ISSUE #5852: etcd 3.0.1 startup failure on Windows 7 64-bit
The following error is seen when trying to start etcd 3.0.1 on Windows 7 64-bit if the `default.etcd` directory tree is not already present:

```
2016-07-02 01:12:50.368813 I | etcdmain: etcd Version: 3.0.1
2016-07-02 01:12:50.368813 I | etcdmain: Git SHA: a4a52cb
2016-07-02 01:12:50.368813 I | etcdmain: Go Version: go1.6.2
2016-07-02 01:12:50.368813 I | etcdmain: Go OS/Arch: windows/amd64
2016-07-02 01:12:50.368813 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
2016-07-02 01:12:50.368813 W | etcdmain: no data-dir provided, using default data-dir ./default.etcd
2016-07-02 01:12:50.368813 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2016-07-02 01:12:50.368813 I | etcdmain: listening for peers on http://localhost:2380
2016-07-02 01:12:50.368813 I | etcdmain: listening for client requests on localhost:2379
2016-07-02 01:12:50.368813 I | etcdserver: name = default
2016-07-02 01:12:50.368813 I | etcdserver: data dir = default.etcd
2016-07-02 01:12:50.368813 I | etcdserver: member dir = default.etcd/member
2016-07-02 01:12:50.368813 I | etcdserver: heartbeat = 100ms
2016-07-02 01:12:50.368813 I | etcdserver: election = 1000ms
2016-07-02 01:12:50.368813 I | etcdserver: snapshot count = 10000
2016-07-02 01:12:50.368813 I | etcdserver: advertise client URLs = http://localhost:2379
2016-07-02 01:12:50.368813 I | etcdserver: initial advertise peer URLs = http://localhost:2380
2016-07-02 01:12:50.368813 I | etcdserver: initial cluster = default=http://localhost:2380
2016-07-02 01:12:50.378813 C | etcdserver: create wal error: rename default.etcd/member/wal.tmp default.etcd/member/wa: Access is denied.
```

This will repeatedly fail until you manually rename `default.etcd/member/wal.tmp` to `default.etcd/member/wal`. Then etcd will start normally.

Notably, this problem is **not** seen in etcd 2.3.7 under the same initial conditions.

### PR DESCRIPTION
**Pull request by heyitsanthony**
Brokenness of master on windows confirmed; works with patch.

Fixes #5852

/cc @xiang90 @MartyMacGyver

### CODE REVIEW

File: wal/wal.go [hunk 1]
```diff
@@ -129,15 +129,22 @@ func Create(dirpath string, metadata []byte) (*WAL, error) {
...
+	newWAL, oerr := Open(dirpath, walpb.Snapshot{})
+	if oerr != nil {
+		return nil, oerr
+	}
+	if _, _, _, err := newWAL.ReadAll(); err != nil {
+		newWAL.Close()
 		return nil, err
 	}
-
-	w.fp = newFilePipeline(w.dir, segmentSizeBytes)
```
Comment (gyuho) [inline wal/wal.go]: why do we delete this?

Comment (heyitsanthony) [inline wal/wal.go]: `Open` will create the file pipeline

Comment (gyuho) [inline wal/wal.go]: Right. thanks.

**Comment by MartyMacGyver** [comment]
Thank you! I rebuilt coreos:master from source with this change in it, and it worked for me as well.

Question: Other than cloning etc with your change and running build.bat, did you go through any other steps before building this on Windows? I've been looking for documentation on building via build.bat and haven't found anything (it clearly works, with some spurious "The directory name is invalid." build initialization errors that don't appear to affect the build itself).

**Comment by xiang90** [comment]
lgtm

**Comment by heyitsanthony** [comment]
@MartyMacGyver just `build.bat` should do it. Please open an issue for the error messages? Thanks!

**Comment by MartyMacGyver** [comment]
@heyitsanthony I'll investigate and then open an issue or two (the how-to-build docs and release notes could use some enhancements in regard to the right way to build and use this in Windows, including prerequisites (e.g., probably need NTFS if you expect to re-create symlinks with Junction calls).

Thanks again for the quick turnaround on this bug!

**Pull request merged**